presto-yarn-test/src/test/groovy/com/teradata/presto/yarn/PrestoClusterTest.groovy

```diff
@@ -187,6 +189,31 @@ class PrestoClusterTest
     }
   }
 
+  @Test
+  void 'flex set of workers - multinode-flex-worker'()
+  {
+    PrestoCluster prestoCluster = new PrestoCluster(slider, hdfsClient, 'resources-multinode-single-worker.json', TEMPLATE)
+    prestoCluster.withPrestoCluster {
+      prestoCluster.assertThatPrestoIsUpAndRunning(1)
+      assertThatAllProcessesAreRunning(prestoCluster)
+      def queryExecutor = prestoCluster.queryExecutor
+      waitForNodesToBeActive(queryExecutor, prestoCluster)
+
+      checkState(workers.size() >= 2, "Number of slaves set in the test yaml configuration should be atleast 3")
+      prestoCluster.flex(WORKER_COMPONENT, 2)
+      waitForNodesToBeActive(queryExecutor, prestoCluster)
+      prestoCluster.assertThatPrestoIsUpAndRunning(2)
+      assertThatAllProcessesAreRunning(prestoCluster)
+
+      prestoCluster.flex(WORKER_COMPONENT, 1)
+      waitForNodesToBeActive(queryExecutor, prestoCluster)
+      prestoCluster.assertThatPrestoIsUpAndRunning(1)
+      assertThatAllProcessesAreRunning(prestoCluster)
```

Comment:
this block of code is similar to the one above, maybe you could wrap this is as method?